src/mlpack/methods/decision_tree/all_categorical_split_impl.hpp

```diff
@@ -209,20 +210,20 @@ double AllCategoricalSplit<FitnessFunction>::SplitIfBetter(
 
 template<typename FitnessFunction>
 size_t AllCategoricalSplit<FitnessFunction>::NumChildren(
-    const double& splitInfo,
+    const arma::vec& splitInfo,
     const AuxiliarySplitInfo& /* aux */)
 {
-  return (size_t) splitInfo;
+  return splitInfo.n_elem == 0 ? 0 : (size_t) splitInfo[0];
```

Comment:
There were  a few small changes that I made along the way in the other split types, and `DecisionTree`, because I needed to change the usage of `splitInfo` for my implementation to work.